### PR TITLE
cluster_claim: always delete claim

### PR DESCRIPTION
**Pull request by AlexNPavel**
The current code immediately exits with an error if the `acquireClaim`
function fails without deleting the created claim. This PR makes the
`acquireClaim` function always return the claim if one is made, even on
error, and the `releaseCluster` function is run even when `aquireClaim`
fails.

/cc @hongkailiu

### CODE REVIEW

File: pkg/steps/cluster_claim.go [hunk 1]
```diff
@@ -63,7 +63,13 @@ func (s *clusterClaimStep) run(ctx context.Context) error {
 	}
 	clusterClaim, err := acquireCluster(ctx, *s.clusterClaim, s.hiveClient, s.client, *s.jobSpec)
 	if err != nil {
-		return results.ForReason("acquiring_cluster_claim").ForError(err)
+		acquireErr := results.ForReason("acquiring_cluster_claim").ForError(err)
+		// always attempt to delete claim if one exists
+		var releaseErr error
+		if clusterClaim != nil {
```
Comment (hongkailiu) [inline pkg/steps/cluster_claim.go:69]: Maybe push line76 (the logging line) to the function `releaseCluster` so that we can see it in both cases of release?

File: pkg/steps/cluster_claim.go [hunk 2]
```diff
@@ -130,32 +136,32 @@ func acquireCluster(ctx context.Context, clusterClaim api.ClusterClaim, hiveClie
 		}
 		return false, nil
 	}); err != nil {
-		return nil, fmt.Errorf("failed to wait for created cluster claim to become running: %w", err)
+		return claim, fmt.Errorf("failed to wait for created cluster claim to become running: %w", err)
```
Comment (hongkailiu) [inline pkg/steps/cluster_claim.go:139]: Add a test for this? Checking it is not nil at error?

**Review by hongkailiu — APPROVED** [review]
